src/big_core_input.cpp

```diff
@@ -112,13 +119,15 @@ const T& BigCoreInput::at(uint64_t imageNum, uint64_t row, uint64_t column, uint
 	if (tileNum >= this->numberOfTiles)
 		throw "Tile number out of bound!";
 
+	int64_t offset = this->outermostEntitiesOffsets[imageNum];
+
 	uint64_t dataTypeIndex = this->isUniformDataType() ? 0 : imageNum;
 	if (this->isInMemory())
 	{
-		return static_cast<const T>(this->data[this->outermostEntitiesOffsets[imageNum] + row * column * plane * tileNum * this->getImageType(this->dataType[dataTypeIndex])]);
+		return static_cast<const T>(this->data[offset + row * column * plane * tileNum * this->getImageType(this->dataType[dataTypeIndex])]);
```

Comment:
Promysli si ještě to indexování. Počítat row * column * plane * tileNum je prostě blbost. Představ si to pro jednoduchost ve 2D matici. Mějme například matici 2 x 3 (height x width):
1 2 3
4 5 6
Matice může být v paměti uložena buď s dataOrder 3,4. Pak to bude v paměti takto: 1 2 3 4 5 6. Nebo s dataOrder 4,3. Pak to bude takto: 1 4 2 5 3 6. Pokud chceme prvek na druhém řádku (row = 1) a ve druhém sloupci (column = 1), musíme v prvním případě počítat index takto: row * width + column = 1 * 3 + 1 = 4. Na tomto indexu je číslo 5. Ve druhém případě počítáme index takto: column * height + row = 1 * 2 + 1 = 3. Na tomto indexu je číslo 5. Takže nám to krásně funguje. No a teď už si to jen představ v 5D místo 2D :-)